src/node/interfaces.cpp

```diff
@@ -754,6 +779,53 @@ class ChainImpl : public Chain
         notifications->blockConnected(block_info);
         if (!block_info.chain_tip && !checkBlocks(locator_block)) return nullptr;
         auto handler = std::make_unique<NotificationsHandlerImpl>(notifications);
+        assert(!handler->m_thread_sync.joinable());
+        if (!block_info.chain_tip) handler->m_thread_sync = std::thread(&util::TraceThread, options.thread_name, [&active, locator_block, notifications, handler = handler.get()] {
```

Comment:
1091e79d7bb57da3b96814fe26101e0bf75e1108 (Move sync thread from index to node)

I think it would be good to abort early and not start the sync process if `CustomInit` of the initial `blockConnected` call failed (because the index might be corrupted). This is the behavior on master, and otherwise the syncing logic could commit some work like syncing another block, and while we'd still terminate with an InitError because `m_init_result=false` after that, this error might no longer occur at the next startup - the index might now still be silently corrupted.